tools/ci_build/github/azure-pipelines/templates/mac-cpu-packing-jobs.yml

```diff
@@ -52,6 +52,19 @@ jobs:
     inputs:
       versionSpec: '18.x'
 
+  - ${{ if eq(parameters.MacosArch, 'x86_64') }}:
```

Comment:
I believe the host architecture is that of macOS-latest, which is not arm64. I think the MacosArch parameter specifies the target architecture.

as this is specifying the Java on the host, could we always use x64?

Reply:
Maybe, provided you didn't want to run the tests, as those will have an arm64 native binary in them and the x64 JVM won't load that. But I'm not clear on how the macOS arm64 support works in the CI pipelines.

Reply:
Is running the tests possible, say for arm64 binaries on an x64 host?

FWIW, I think we only run tests when MacosArch is x86_64:
https://github.com/microsoft/onnxruntime/blob/b95fd4e644775a4343c13435bd729bd64f411752/tools/ci_build/github/azure-pipelines/templates/mac-cpu-packaging-steps.yml#L55-L58

AFAIK, none of the Mac CI agents we use now (all hosted agents) are arm64.

Reply:
I don't think it's possible to run the arm64 tests on an x64 host without some kind of VM infrastructure. So I probably should tell it to load the x64 JVM everywhere. Presumably there's something in the CMake which forces the binaries to build as arm64 on x64, but nothing actually executes those binaries?

Reply:
> Presumably there's something in the CMake which forces the binaries to build as arm64 on x64, but nothing actually executes those binaries?

Right, we can cross-compile. It looks like the build definition is skipping running of the tests for arm64.

The latest MacOS CI build succeeded. I think we can simplify this to just use the x64 JVM and it should be good.

Reply:
Done.